sway-parse/src/attribute.rs

```diff
@@ -1,13 +1,78 @@
+use crate::priv_prelude::{Peek, Peeker};
 use crate::{Parse, ParseBracket, ParseErrorKind, ParseResult, ParseToEnd, Parser, ParserConsumed};
 
 use sway_ast::attribute::{Annotated, Attribute, AttributeDecl};
-use sway_ast::brackets::Parens;
-use sway_ast::keywords::{HashToken, StorageToken};
-use sway_types::Ident;
+use sway_ast::brackets::{Parens, SquareBrackets};
+use sway_ast::keywords::{CommaToken, HashToken, StorageToken, Token};
+use sway_ast::punctuated::Punctuated;
+use sway_ast::token::{DocComment, DocStyle};
+use sway_types::{Ident, Span};
+
+impl Peek for DocComment {
+    fn peek(peeker: Peeker<'_>) -> Option<DocComment> {
+        peeker
+            .peek_doc_comment()
+            .ok()
+            .map(|doc_comment| doc_comment.clone())
+        // .map(DocComment::clone)
```

Comment:
`.map(DocComment::clone)` did not work, but Clippy still wants me to do it. Need to have a look.